src/main/java/dbProcs/Getter.java

```diff
@@ -119,59 +128,66 @@
 			log.debug("Getting password hash");
 			String dbHash;
 			try {
-				dbHash = userFind.getString(3);
+				dbHash = userResult.getString(3);
 				log.debug("Verifying hash");
 
 				userVerified = argon2.verify(dbHash, password.toCharArray());
 
 			} catch (SQLException e) {
-				log.error("Could not retrieve password hash from db: " + e.toString());
+				log.fatal("Could not retrieve password hash from db: " + e.toString());
 				result = null;
-				userVerified=false;
+				userVerified = false;
+				throw new RuntimeException(e);
 				// TODO: We should throw a checked exception here instead
 			}
 
 			if (userVerified) {
 				// Hash matches
 				log.debug("Hash matches");
 
-				result = new String[5];
-				boolean isTempPassword;
+				result = new String[6];
+
 				int badLoginCount;
-				
+				String loginType = new String();
+
 				Timestamp suspendedUntil;
 
 				try {
-					result[0] = userFind.getString(1);
-					result[1] = userFind.getString(2); // userName
-					result[2] = userFind.getString(4); // role
-					badLoginCount=userFind.getInt(5);
-					isTempPassword = userFind.getBoolean(6);
-					result[4] = userFind.getString(7); // classId
-					suspendedUntil = userFind.getTimestamp(8);
+					result[0] = userResult.getString(1);
+					result[1] = userResult.getString(2); // userName
+					result[2] = userResult.getString(4); // role
+					badLoginCount = userResult.getInt(5);
+					result[3] = Boolean.toString(userResult.getBoolean(6));
+					result[4] = userResult.getString(7); // classId
+					suspendedUntil = userResult.getTimestamp(8);
+					loginType = userResult.getString(9);
+					result[5] = Boolean.toString(userResult.getBoolean(10));
 				} catch (SQLException e) {
+
 					log.fatal("Could not retrieve auth data from db: " + e.toString());
 					throw new RuntimeException(e);
-				} 
+				}
+
+				if (!loginType.equals("login")) {
+					// Login type must be "login" and not "saml" if password login is to be allowed
+					log.debug("User is SSO user, can't login with password!");
```

Comment:
It appears so, I must have been sloppy.

I have queued up a lot of PR:s now, and they all build on each other. Perhaps we should review the very latest PR, it will make a bit more sense then :)

Reply:
Are the changes in this PR included in a later PR? Should this PR be closed? 

It takes a bit of time to review these PR's because of all the formatting churn.